### PR TITLE
Tca 28 implement unit test for the splash section

### PR DESCRIPTION
**Pull request by pscacchi**
UnitTest was created.
These tests check the spashviewmodel and the splashtimerusecase

### CODE REVIEW

File: app/src/test/java/ar/scacchipa/twittercloneapp/domain/SpahsTimerUseCaseUnitTest.kt [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import org.junit.Test
+
+@ExperimentalCoroutinesApi
+class SpahsTimerUseCaseUnitTest {
+    @Test
+    fun returnTrueFiveSecondLater() = runTest {
+        var wasSpent = false
+        val timer = SplashTimerUseCase()
+        assertEquals(false, wasSpent)
+        this.runCurrent()
```
Comment (sebastianMullerEndava) [inline app/src/test/java/ar/scacchipa/twittercloneapp/domain/SpahsTimerUseCaseUnitTest.kt:18]: I think this is not necessary

File: app/src/test/java/ar/scacchipa/twittercloneapp/domain/SpahsTimerUseCaseUnitTest.kt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+            wasSpent = timer.spendSplash()
+        }
+        assertEquals(false, wasSpent)
+        val initialTime = this.testScheduler.currentTime
+        this.advanceUntilIdle()
+        val changeTime = this.testScheduler.currentTime
+
+        assertEquals(5000, changeTime - initialTime)
+        assertEquals(true, wasSpent)
+        this.advanceUntilIdle()
```
Comment (sebastianMullerEndava) [inline app/src/test/java/ar/scacchipa/twittercloneapp/domain/SpahsTimerUseCaseUnitTest.kt:29]: same here

File: app/src/test/java/ar/scacchipa/twittercloneapp/domain/SpahsTimerUseCaseUnitTest.kt (hunk 1 above)
Comment (pscacchi) [reply app/src/test/java/ar/scacchipa/twittercloneapp/domain/SpahsTimerUseCaseUnitTest.kt:18]: The this.runCurrent() was deleted.

File: app/src/test/java/ar/scacchipa/twittercloneapp/domain/SpahsTimerUseCaseUnitTest.kt (hunk 2 above)
Comment (pscacchi) [reply app/src/test/java/ar/scacchipa/twittercloneapp/domain/SpahsTimerUseCaseUnitTest.kt:29]: the lines 29-30 was deleted

File: app/src/androidTest/java/ar/scacchipa/twittercloneapp/ui/ExampleInstrumentedTest.kt [hunk 3]
```diff
@@ -1,24 +1,42 @@
...
-        val appContext = InstrumentationRegistry.getInstrumentation().targetContext
-        assertEquals("ar.scacchipa.twittercloneapp", appContext.packageName)
+    fun splashSpendInFiveSeconds() {
+        onView(withId(R.id.twitter_logo_imageview))
+            .check(matches(isDisplayed()))
+
+        onView(withId(R.layout.fragment_login_layout))
+            .check(doesNotExist())
+
+        Thread.sleep(4000)
```
[the remark this answers is not in the document]
Comment (pscacchi) [reply app/src/androidTest/java/ar/scacchipa/twittercloneapp/ui/ExampleInstrumentedTest.kt:32]: That's right.

File: app/src/test/java/ar/scacchipa/twittercloneapp/domain/SpahsTimerUseCaseUnitTest.kt [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+class SpahsTimerUseCaseUnitTest {
+    @Test
+    fun returnTrueFiveSecondLater() = runTest {
+        var wasSpent = false
+        val timer = SplashTimerUseCase()
+        assertEquals(false, wasSpent)
+        val initialTime = this.testScheduler.currentTime
+        wasSpent = timer.spendSplash()
+
+        val changeTime = this.testScheduler.currentTime
```
Comment (sebastianMullerEndava) [inline app/src/test/java/ar/scacchipa/twittercloneapp/domain/SpahsTimerUseCaseUnitTest.kt:18]: no need to play with the scheduler I think.

**Review by Diana-Endava — APPROVED** [review]

**Review by sebastianMullerEndava — APPROVED** [review]

**Pull request merged**